infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServerCache.java

```diff
@@ -126,85 +101,127 @@ public void delete(@Nonnull ConflictTxStream key, Long globalAddress, @Nonnull R
-     * Put a value in the cache
+     * The cache size as the number of entries
      *
-     * @param conflictStream conflict stream
-     * @param newTail        global tail
+     * @return cache size
      */
-    public void put(ConflictTxStream conflictStream, long newTail) {
-        conflictCache.put(conflictStream, newTail);
+    public int size() {
+        return cacheConflictKeys.size();
     }
 
     /**
-     * Discard all entries in the cache
+     * The memory space used by the entries and also the space used by
+     * priorityque, hashmap to store the pointers
+     * @return the memory space used in bytes:
      */
-    public void invalidateAll() {
-        log.info("Invalidate sequencer server cache");
-        conflictCache.invalidateAll();
+    public long byteSize() {
+        return cacheEntriesBytes + this.size()*ENTRY_OVERHEAD;
     }
 
-    /**
-     * Update max conflict wildcard by a new address
+    /*
+     * Put a value in the cache
      *
-     * @param newMaxConflictWildcard new conflict wildcard
+     * @param conflictStream conflict stream
      */
-    public void updateMaxConflictAddress(long newMaxConflictWildcard) {
-        log.info("updateMaxConflictAddress, new address: {}", newMaxConflictWildcard);
-        maxConflictWildcard = newMaxConflictWildcard;
-        maxConflictNewSequencer = newMaxConflictWildcard;
+    public boolean put(ConflictTxStream conflictStream) {
+
+        Long val = cacheConflictKeys.get(conflictStream);
+        if (val != null && val >= conflictStream.txVersion) {
+            log.error("For key {}, the value {} is not smaller  than the expected value {} or maxWildCard",
+                    conflictStream, cacheConflictKeys.get(conflictStream), conflictStream.txVersion, maxConflictWildcard);
+            return false;
+        }
+
+        if (cacheConflictKeys.size() == cacheSize) {
+            invalidateFirst();
+        }
+
+        cacheEntries.add(conflictStream);
+        cacheConflictKeys.put(conflictStream, conflictStream.txVersion);
+        cacheEntriesBytes += conflictStream.size();
+        return true;
     }
 
     /**
      * Contains the conflict hash code for a stream ID and conflict param.
      */
     @EqualsAndHashCode
     public static class ConflictTxStream {
+
+        @Getter
         private final UUID streamId;
+        @Getter
         private final byte[] conflictParam;
 
-        public ConflictTxStream(UUID streamId, byte[] conflictParam) {
+        @EqualsAndHashCode.Exclude
+        public final long txVersion;
+
+        public ConflictTxStream(UUID streamId, byte[] conflictParam, long address) {
             this.streamId = streamId;
             this.conflictParam = conflictParam;
+            txVersion = address;
         }
 
         @Override
         public String toString() {
             return streamId.toString() + conflictParam;
         }
+
+        public long size() {
+            return MetricsUtils.sizeOf.deepSizeOf(this);
```

Comment:
Why we need to measure the size of the object in memory?
It's a very rare and fragile operation in java to measure the object size. Is it possible to avoid that?
What we actually measuring here and how we are going to use it?

Reply:
Made the change.